File: daily/Binary Tree/145-Binary-tree-postorder-traversal.go
package daily

func postorderTraversal(root *TreeNode) []int {
	var results []int
	dfs(root, &results)
	return results
}

func dfs(node *TreeNode, results *[]int) {
	if node == nil {
		return
	}
	dfs(node.Left, results)
	dfs(node.Right, results)
	*results = append(*results, node.Val)
}

File: daily/Binary Tree/145-Binary-tree-postorder-traversal.py
class Solution:
    def postorderTraversal(self, root: Optional[TreeNode]) -> List[int]:
        results = []
        def dfs(node):
            if not node:
                return
            dfs(node.left)
            dfs(node.right)
            results.append(node.val)
        dfs(root)
        return results